### PR TITLE
Allow skipping snap installation in preinst when lxd is not activated yet

### PR DESCRIPTION
**Pull request by rbalint**
LP: #1821924
Signed-off-by: Balint Reczey <balint.reczey@canonical.com>

### CODE REVIEW

**Comment by rbalint** [comment]
This fixes the case when snap command is not functional for one reason or an other, like when running in Chrome OS, WSL or just behind a firewall that can't be opened easily.

https://forum.snapcraft.io/t/snapd-errors-when-init-isnt-systemd/10684/23
https://forum.snapcraft.io/t/windows-subsystem-for-linux/216

**Comment by rbalint** [comment]
@stgraber I'd happily target the PR against master, but there is no such branch.

File: lxc [hunk 1]
```diff
@@ -1,2 +1,8 @@
 #!/bin/sh
+if ! [ -x /snap/bin/lxc ]; then
+    echo "Command '$0' requires the lxd snap to be installed." >2
+    echo "Please install it with:" >2
+    echo "" >2
+    echo "snap install lxd  # version 3.11" >2
```
Comment (stgraber) [inline lxc]: What's that `# version 3.11` comment here? We release monthly so I don't think we should include that anywhere in the deb.

File: lxd [hunk 2]
```diff
@@ -1,2 +1,8 @@
 #!/bin/sh
+if ! [ -x /snap/bin/lxd ]; then
+    echo "Command '$0' requires the lxd snap to be installed." >2
+    echo "Please install it with:" >2
+    echo "" >2
+    echo "snap install lxd  # version 3.11" >2
```
Comment (stgraber) [inline lxd]: What's that `# version 3.11` comment here? We release monthly so I don't think we should include that anywhere in the deb.

File: debian/lxd.preinst [hunk 3]
```diff
@@ -41,6 +42,14 @@ case "$1" in
                 if [ "${RET}" = "Abort" ]; then
                     echo "===> Aborting at user request"
                     exit 1
+                elif [ "${RET}" = "Skip" ]; then
+                    if ! [ -e "/var/lib/lxd/server.crt" ]; then
+                        echo "===> Skipping at user request"
+                        SKIP=true
+                        break
+                    else
+                        echo "===> Skipping is not allowed, lxd is activated on the system"
```
Comment (stgraber) [inline debian/lxd.preinst]: This should be shown as a debconf dialog if possible too.

File: debian/lxd.preinst [hunk 4]
```diff
@@ -51,48 +60,53 @@ case "$1" in
             COUNT=$((COUNT+1))
         done
 
-        # Set the default track
-        db_get lxd/snap-track
-        if [ -z "${RET}" ]; then
-            if [ -e "/etc/os-release" ] && grep -q " LTS " /etc/os-release; then
-                db_set lxd/snap-track "3.0"
+        if ! $SKIP; then
```
Comment (stgraber) [inline debian/lxd.preinst]: Use `${SKIP}` for consistency?

**Comment by stgraber** [comment]
`dpm-disco` is now available, please re-submit to that when you have the changes ready.

In theory I've reconfigured Jenkins to consider dpm-disco the right branch for this repo.

File: debian/lxd.preinst (hunk 4 above)
Comment (rbalint) [reply debian/lxd.preinst]: Fixed all of those in https://github.com/lxc/lxd-pkg-ubuntu/pull/34

**Pull request closed without merging**